### PR TITLE
fix: migrate fullscreen store and related functions/listeners to composable

### PR DESCRIPTION
**Pull request by Antreesy**
### 🖼️ Screenshots / Screencasts

No visual changes

### 🏁 Checklist

- [ ] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [ ] Talk Desktop
  - [ ] Not risky to browser differences / client
- [ ] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/composables/useDocumentFullscreen.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	})
+
+	return readonly(isFullscreen)
+}
+
+/**
+ * Enable a fullscreen with Fullscreen API
+ */
+export async function enableFullscreen() {
+	const element = document.getElementById('content-vue')
```
Comment (DorraJaouad) [inline src/composables/useDocumentFullscreen.ts:42]:
```suggestion
	const element = document.getElementById('content-vue')
	
	if (!element) {
		return
	}
```

File: src/composables/useDocumentFullscreen.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	} else if (element.msRequestFullscreen) {
+		await element.msRequestFullscreen()
+	}
+}
+
+/**
+ * Disable a fullscreen
+ */
+export async function disableFullscreen() {
+	if (document.exitFullscreen) {
```
Comment (DorraJaouad) [inline src/composables/useDocumentFullscreen.ts:59]:
```suggestion
	if (document?.exitFullscreen) {
```

File: src/composables/useDocumentFullscreen.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	const changeIsFullscreen = () => {
+		isFullscreen.value = document.fullscreenElement !== null
+	}
+
+	onBeforeMount(() => {
+		document.addEventListener('fullscreenchange', changeIsFullscreen)
+		document.addEventListener('mozfullscreenchange', changeIsFullscreen)
+		document.addEventListener('MSFullscreenChange', changeIsFullscreen)
+		document.addEventListener('webkitfullscreenchange', changeIsFullscreen)
+	})
```
Comment (ShGKme) [inline src/composables/useDocumentFullscreen.ts:26]: Same as
```suggestion
document.addEventListener('fullscreenchange', changeIsFullscreen)
document.addEventListener('mozfullscreenchange', changeIsFullscreen)
document.addEventListener('MSFullscreenChange', changeIsFullscreen)
document.addEventListener('webkitfullscreenchange', changeIsFullscreen)
```

File: src/composables/useDocumentFullscreen.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+	if (element.requestFullscreen) {
+		await element.requestFullscreen()
+	} else if (element.webkitRequestFullscreen) {
+		await element.webkitRequestFullscreen(Element.ALLOW_KEYBOARD_INPUT)
+	} else if (element.mozRequestFullScreen) {
+		await element.mozRequestFullScreen()
+	} else if (element.msRequestFullscreen) {
+		await element.msRequestFullscreen()
+	}
```
Comment (ShGKme) [inline src/composables/useDocumentFullscreen.ts:52]: We only need `webkit` prefix for Safari now

https://caniuse.com/?search=requestFullscreen

File: src/composables/useDocumentFullscreen.ts (hunk 2 above)
Comment (ShGKme) [reply src/composables/useDocumentFullscreen.ts:59]: `document` should be defined. We don't have SSR and we won't use this composable in a worker

File: src/composables/useDocumentFullscreen.ts (hunk 1 above)
Comment (ShGKme) [inline src/composables/useDocumentFullscreen.ts:42]: In the current implementation, it enabled Fullscreen specifically on `#content-vue`, not a general fullscreen

**Review by ShGKme — APPROVED** [review]

File: src/composables/useDocumentFullscreen.ts (hunk 1 above)
Comment (Antreesy) [reply src/composables/useDocumentFullscreen.ts:42]: Will be changed in follow-up

**Review by ShGKme — APPROVED** [review]

**Pull request merged**